modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/properties/time/TimestampFileNameExtractorSPI.java

```diff
@@ -71,38 +79,36 @@ public PropertiesCollector create(final Object o, final List<String> propertyNam
 
                 String value = (String) o;
 
-                // look for the regex
-                int idx = 0;
-                if (value.startsWith(REGEX_PREFIX)) {
-                    String tmp = value.substring(REGEX_PREFIX.length());
-                    if (tmp.contains("," + FORMAT_PREFIX)) {
-                        idx = tmp.indexOf("," + FORMAT_PREFIX);
-                        regex = tmp.substring(0, idx);
-                        value = tmp.substring(idx + 1);
-                    } else if (tmp.contains("," + FULL_PATH_PREFIX)) {
-                        idx = tmp.indexOf("," + FULL_PATH_PREFIX);
-                        regex = tmp.substring(0, idx);
-                        value = tmp.substring(idx + 1);
-                    } else {
-                        regex = tmp;
-                    }
-                }
+                int minIndex = value.length();
```

Comment:
I have no clue what's going on in this block of changes :-D

Reply:
Eh :)
that block was looking for parameters (format, fullPath, regex itself) in the parser definition using a convoluted set of if-s and string find.
I have been refactored that by simply putting all of these parameters into a Properties object so it's easier to find them no matter how they are sorted in the String.